apps/hash-api/src/graph/knowledge/primitive/entity.ts

```diff
@@ -717,6 +723,74 @@ export const canUpdateEntity: ImpureGraphFunction<
     .canUpdateEntity(actorId, params.entityId)
     .then(({ data }) => data.has_permission);
 
+export const checkPermissionsOnEntity: ImpureGraphFunction<
+  { entity: Pick<Entity, "metadata"> },
+  Promise<UserPermissions>
+> = async (graphContext, { actorId }, params) => {
+  const { entity } = params;
+
+  const { entityId } = entity.metadata.recordId;
+
+  const isAccountGroup =
+    entity.metadata.entityTypeId === SYSTEM_TYPES.entityType.org.schema.$id;
+
+  const isPublicUser = actorId === publicUserAccountId;
+
+  const [entityEditable, membersEditable] = await Promise.all([
+    isPublicUser
+      ? false
+      : await canUpdateEntity(graphContext, { actorId }, { entityId }),
+    isAccountGroup
+      ? isPublicUser
+        ? false
+        : await graphContext.graphApi
+            .canAddGroupMember(actorId, extractEntityUuidFromEntityId(entityId))
+            .then(({ data }) => data.has_permission)
+      : null,
+  ]);
```

Comment:
To simplify this I like to extend the current reading-interface to allow specifying filters. Things I can filter on the graph:
- entity id
- relation between entity and subject
- subject type (i.e. `account`, `accountGroup` or `public`)
- subject id (i.e. `accountId`, `accountGroupId`)
- subject relation (i.e. `member` for account groups)

If this would help I can do this as one of my next tasks.

Reply:
Oh, that's true. I think there is an API endpoint on spiceDB to do something like that but I wouldn't look into that now. Sorry for the noise